base_layer/core/src/block.rs

```diff
@@ -44,3 +44,72 @@ pub struct AggregateBody {
     /// Kernels contain the excesses and their signatures for transaction
     pub kernels: Vec<TransactionKernel>,
 }
+
+impl AggregateBody {
+    /// Create an empty aggregate body
+    pub fn empty() -> AggregateBody {
+        AggregateBody { inputs: vec![], outputs: vec![], kernels: vec![] }
+    }
+
+    /// Create a new aggregate body from provided inputs, outputs and kernels
+    pub fn new(
+        inputs: Vec<TransactionInput>,
+        outputs: Vec<TransactionOutput>,
+        kernels: Vec<TransactionKernel>,
+    ) -> AggregateBody
+    {
+        AggregateBody { inputs, outputs, kernels }
+    }
+
+    /// Add an input to the existing aggregate body
+    pub fn with_input(self, input: TransactionInput) -> AggregateBody {
+        let mut new_inputs = self.inputs;
+        new_inputs.push(input);
+        new_inputs.sort();
+        AggregateBody { inputs: new_inputs, ..self }
+    }
+
+    /// Add a series of inputs to the existing aggregate body
```

Comment:
add comment that this destroys vec inputs

Reply:
I feel like the function header makes that clear?